client/cmd/flags.go

```diff
@@ -377,3 +647,83 @@ func validateCommissionRate(ctx context.Context, cfg *createValidatorConfig) err
 
 	return nil
 }
+
+func validateNewCommissionRate(ctx context.Context, cfg *updateCommissionConfig, pubKey []byte) error {
+	validator, err := getValidatorByEVMAddr(ctx, cfg.API, pubKey)
+	if err != nil {
+		return err
+	}
+
+	if validator.OperatorAddress == "" {
+		return errors.New("validator not found")
+	}
+
+	prevCommission := validator.Commission
+	newCommission := math.LegacyNewDecWithPrec(int64(cfg.CommissionRate), 4)
+
+	if time.Since(prevCommission.UpdateTime).Hours() < 24 {
+		return stypes.ErrCommissionUpdateTime
+	}
+
+	if newCommission.GT(prevCommission.MaxRate) {
+		return stypes.ErrCommissionGTMaxRate
+	}
+
+	if newCommission.Sub(prevCommission.Rate).GT(prevCommission.MaxChangeRate) {
+		return stypes.ErrCommissionGTMaxChangeRate
+	}
+
+	return nil
+}
+
+type ValidatorResponse struct {
+	Code  int    `json:"code"`
+	Error string `json:"error"`
+	Msg   struct {
+		Validator stypes.Validator `json:"validator"`
+	} `json:"msg"`
+}
+
+// isValidatorFound checks whether a validator with the given public key exists.
+func isValidatorFound(ctx context.Context, endpoint string, pubKey []byte) (bool, error) {
+	validator, err := getValidatorByEVMAddr(ctx, endpoint, pubKey)
+	if err != nil {
+		return false, err
+	}
+
+	return validator.OperatorAddress != "", nil
+}
+
+// getValidatorByEVMAddr gets a validator with the given public key.
+func getValidatorByEVMAddr(ctx context.Context, endpoint string, pubKey []byte) (stypes.Validator, error) {
+	valEVMAddr, err := k1util.CosmosPubkeyToEVMAddress(pubKey)
+	if err != nil {
+		return stypes.Validator{}, errors.Wrap(err, "failed to convert pub key to evm address")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("%s/staking/validators/%s", endpoint, valEVMAddr), nil)
+	if err != nil {
+		return stypes.Validator{}, errors.Wrap(err, "failed to create request for getting validator")
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return stypes.Validator{}, errors.Wrap(err, "failed to get validator")
+	}
```

Comment:
If the request fails (wrong API value), instead of failing, we could emit a warning about failure to validate and proceed to execute the requested action (e.g. update commission).

Reply:
User can skip this validation by not using api flag, so strict validation would be good even if the request fails with api flag. wdyt?

Reply:
sounds good